### ISSUE TITLE
Missing lines in Readme examples?

### ISSUE THREAD

**Issue by nicolas-ivanov** [huggingface/pytorch-transformers#944]
1. In the [example](https://github.com/huggingface/pytorch-transformers#serialization)
```
...
### Do some stuff to our model and tokenizer
# Ex: add new tokens to the vocabulary and embeddings of our model
tokenizer.add_tokens(['[SPECIAL_TOKEN_1]', '[SPECIAL_TOKEN_2]'])
model.resize_token_embeddings(len(tokenizer))
# Train our model
train(model)
...
```

`model.train()` is missing before `train(model)` ?

2. In the [example](https://github.com/huggingface/pytorch-transformers#optimizers-bertadam--openaiadam-are-now-adamw-schedules-are-standard-pytorch-schedules)
```
...
### In PyTorch-Transformers, optimizer and schedules are splitted and instantiated like this:
optimizer = AdamW(model.parameters(), lr=lr, correct_bias=False)  # To reproduce BertAdam specific behavior set correct_bias=False
scheduler = WarmupLinearSchedule(optimizer, warmup_steps=num_warmup_steps, t_total=num_total_steps)  # PyTorch scheduler
### and used like this:
for batch in train_data:
    loss = model(batch)
    loss.backward()
    scheduler.step()
    optimizer.step()
```

`optimizer.zero_grad()` is missing after `optimizer.step()` ?

**Comment by thomwolf** [comment]
Thanks

**Comment by nicolas-ivanov** [comment]
@thomwolf note: the first example hasn't been fixed by the commit.

**Comment by thomwolf** [comment]
Yes, doesn't look like a problem to me. Usually, people put the model in training mode inside the train function (and even inside the training loop I would recommend).

**Comment by nicolas-ivanov** [comment]
Ok, got it!
